core/plannernotes.c

```diff
@@ -49,10 +52,11 @@ void add_plan_to_notes(struct diveplan *diveplan, struct dive *dive, bool show_d
 	bool gaschange_after = !plan_verbatim;
 	bool gaschange_before;
 	bool lastentered = true;
+	bool istrimix = false;
```

Comment:
Maybe it would be a good idea to move all these frees (for early return) to the end of the function so they occur only once and jump to it via some goto finished label.